packages/govuk-frontend/src/govuk/components/accordion/accordion.mjs

```diff
@@ -394,18 +410,30 @@ export class Accordion extends GOVUKFrontendComponent {
     const $showHideText = $section.querySelector(
       `.${this.sectionShowHideTextClass}`
     )
+
     const $button = $section.querySelector(`.${this.sectionButtonClass}`)
     const $content = $section.querySelector(`.${this.sectionContentClass}`)
 
-    if (
-      !$showHideIcon ||
-      !($showHideText instanceof HTMLElement) ||
-      !$button ||
-      !$content
-    ) {
+    // Return early for elements created during set up
+    if (!$showHideIcon || !$showHideText) {
       return
     }
 
+    if (!$button || !$content) {
+      const [check] = /** @type {const} */ ([
+        [$button, this.sectionButtonClass],
+        [$content, this.sectionContentClass]
```

Comment:
Alternatively, with only two selectors, is it worth having an array + filter?

Was a better idea when we had 4x elements

Reply:
Yeah, this is a great point. I'll do some rejigging.

Reply:
I've split this out into a check for `$button`, a check for `$content`, and a check for the two internally created classes.

It's a similar amount of code, and I think makes things super easy to parse.